src/node_perf.cc

```diff
@@ -373,6 +374,47 @@ void Timerify(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(wrap);
 }
 
+static void on_uv_interval(uv_timer_t* req) {
+  delay_timer_t* delay = reinterpret_cast<delay_timer_t*>(req->data);
+
+  uint64_t time = uv_hrtime();
+  uint64_t delta = time - delay->prev;
+
+  if (delta > DELAY_TIMER_THRESHOLD_NS) {
+    delta = (delta - DELAY_TIMER_RESOLUTION_NS) / 1e6;
+    if (delta > 0xffffffff) delta = 0xffffffff;
+    delay->delay += (uint32_t) delta;
+    delay->times++;
+    TRACE_COUNTER1(TRACING_CATEGORY_NODE2(perf, event_loop),
+                   "event loop delay",
+                   delay->delay);
+  }
+
+  delay->prev = time;
+}
+
+void StartEventLoopTimer(const FunctionCallbackInfo<Value>& args) {
+  CHECK(Buffer::HasInstance(args[0]));
+  delay_timer_t* delay =
+      reinterpret_cast<delay_timer_t*>(node::Buffer::Data(args[0]));
+
+  uv_timer_init(uv_default_loop(), &(delay->timer));
```

Comment:
Btw, the extra parentheses in `&(delay->timer)` are pretty unidiomatic for our code.